midend/local_copyprop.cpp

```diff
@@ -140,9 +154,20 @@ IR::AssignmentStatement *P4::DoLocalCopyPropagation::postorder(IR::AssignmentSta
     return as;
 }
 
+IR::MethodCallExpression *P4::DoLocalCopyPropagation::postorder(IR::MethodCallExpression *mc) {
+    if (!working) return mc;
+    LOG3("method call " << mc->method << "clears all saved values");
+    /* FIXME -- should look at what the called method actually reads and modifies, rather
+     * than assuming everything */
+    for (auto &var : Values(available)) {
```

Comment:
Yes -- that's why it just goes through and erases all the accumulated value/use state (it assumes the method/table actions might modify any of the variables we've seen.)  Note that effects on parameters have already been dealt with (when we visited the parameters).

Note the common case here is a table.apply method, in which case we need to analyze the bodies of all the actions that might run as a result -- non-trivial.